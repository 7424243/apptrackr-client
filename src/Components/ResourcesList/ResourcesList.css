hr {
    margin-left: 20px;
    margin-right: 20px;
}

ul {
    margin-bottom: 30px;
}

/* Classes */

.resource_item {
    margin-bottom: 20px;
}

.resources_section1, .resources_section2 {
    text-align: left;
}

.resources_section1_content {
    margin-top: 20px;
    margin-left: 20px;
    margin-right: 20px;
    padding: 15px;
}

.resources_section2_content {
    margin-left: 20px;
    margin-right: 20px;
    padding: 15px;
}

/* Media Query */

@media only screen and (min-width: 650px) {

    ul {
        list-style-position: inside
    }

    .resource_item {
        display: flex;
        justify-content: center;
    }
    
    .resources_section1, .resources_section2 {
        text-align: center;
    }

}