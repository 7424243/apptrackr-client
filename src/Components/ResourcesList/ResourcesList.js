import React, {Component} from 'react'
import ResourceItem from '../ResourceItem/ResourceItem'
import SquareButton from '../SquareButton/SquareButton'
import './ResourcesList.css'

class ResourcesList extends Component {
    render() {
        return (
            <div>
                <section>
                    <div>
                        <h2>Job Boards</h2>
                        <ul>
                            <div>
                                <ResourceItem 
                                    name={'Indeed'} 
                                    url={'https://www.indeed.com/'}
                                />
                                <SquareButton 
                                    content={'x'}
                                    path={'/resources'}
                                />
                            </div>
                            <div>
                                <ResourceItem 
                                    name={'LinkedIn'} 
                                    url={'https://www.linkedin.com/jobs/?trk=li_sem_namer_careers_jobsgtm_Careers_NAMER_T1_USCA_Search_Google-Brand_DR-PRS_Multiple_Brand-LinkedIn-Jobs-Exact_Desktop_English_Core_network=g_campaign=11602008719_keyword=linkedin%20jobs&src=go-pa&veh=Careers_NAMER_T1_USCA_Search_Google-Brand_DR-PRS_Multiple_Brand-LinkedIn-Jobs-Exact_Desktop_English_Core_479259659412_linkedin%20jobs_c__kwd-1528692947_11602008719&mcid=&cname=Careers_NAMER_T1_USCA_Search_Google-Brand_DR-PRS_Multiple_Brand-LinkedIn-Jobs-Exact_Desktop_English_Core&camid=11602008719&asid=113158397036&keyword=linkedin%20jobs&targetid=kwd-1528692947&crid=479259659412&placement=&dev=c&gclid=Cj0KCQiA7NKBBhDBARIsAHbXCB7OLVWs8pYmpW31PIvUl-AppU9qdI8QLsUKJMSsiLifTs41no_v3aYaAhwZEALw_wcB'}
                                />
                                <SquareButton 
                                    content={'x'}
                                    path={'/resources'}
                                />
                            </div>
                            <div>
                                <ResourceItem 
                                    name={'Apprentice.at'} 
                                    url={'https://apprentice.at/'}
                                />
                                <SquareButton 
                                    content={'x'}
                                    path={'/resources'}
                                />
                            </div>
                        </ul>
                        <SquareButton 
                            content={'+'}
                            path={'/newresource'}
                        />
                    </div>
                </section>
                <hr/>
                <section>
                    <div>
                        <h2>Resume & Cover Letter Resources</h2>
                        <ul>
                            <div>
                                <ResourceItem 
                                    name={'Resume'} 
                                    url={'https://resumekraft.com/wp-content/uploads/2019/12/teacher-resume-example.jpg'}
                                />
                                <SquareButton 
                                    content={'x'}
                                    path={'/resources'}
                                />
                            </div>
                            <div>
                                <ResourceItem 
                                    name={'Cover Letter'} 
                                    url={'https://www.thebalancecareers.com/thmb/IC37LV1es8xHWzy2TdJi5sFvIbg=/1650x1275/filters:no_upscale():max_bytes(150000):strip_icc()/The_Balance_Cover_2060250Aug-edddc2451f264a6eb1c94cf627cb77ff.jpg'}
                                />
                                <SquareButton 
                                    content={'x'}
                                    path={'/resources'}
                                />
                            </div>
                            <div>
                                <ResourceItem 
                                    name={'Resume Optimizer'} 
                                    url={'https://skillsyncer.com/'}
                                />
                                <SquareButton 
                                    content={'x'}
                                    path={'/resources'}
                                />
                            </div>

                        </ul>
                        <SquareButton 
                            content={'+'}
                            path={'/newresource'}
                        />
                    </div>
                </section>
            </div>
        )
    }
}

export default ResourcesList