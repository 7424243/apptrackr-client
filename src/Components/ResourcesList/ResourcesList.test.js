import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import ResourcesList from './ResourcesList'

describe('ResourcesList Component', () => {

    const props = {
        history: {
            push: jest.fn()
        }
    }

    //smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter><ResourcesList {...props}/></BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    //snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<BrowserRouter><ResourcesList {...props}/></BrowserRouter>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
    
})