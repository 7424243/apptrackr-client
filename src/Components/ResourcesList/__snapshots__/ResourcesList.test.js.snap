// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourcesList Component renders the UI as expected 1`] = `
<div>
  <section>
    <div>
      <h2>
        Job Boards
      </h2>
      <ul>
        <div>
          <li>
            <a
              href="https://www.indeed.com/"
              rel="noreferrer"
              target="_blank"
            >
              Indeed
            </a>
          </li>
          <div>
            <a
              href="/resources"
              onClick={[Function]}
            >
              <button>
                x
              </button>
            </a>
          </div>
        </div>
        <div>
          <li>
            <a
              href="https://www.linkedin.com/jobs/?trk=li_sem_namer_careers_jobsgtm_Careers_NAMER_T1_USCA_Search_Google-Brand_DR-PRS_Multiple_Brand-LinkedIn-Jobs-Exact_Desktop_English_Core_network=g_campaign=11602008719_keyword=linkedin%20jobs&src=go-pa&veh=Careers_NAMER_T1_USCA_Search_Google-Brand_DR-PRS_Multiple_Brand-LinkedIn-Jobs-Exact_Desktop_English_Core_479259659412_linkedin%20jobs_c__kwd-1528692947_11602008719&mcid=&cname=Careers_NAMER_T1_USCA_Search_Google-Brand_DR-PRS_Multiple_Brand-LinkedIn-Jobs-Exact_Desktop_English_Core&camid=11602008719&asid=113158397036&keyword=linkedin%20jobs&targetid=kwd-1528692947&crid=479259659412&placement=&dev=c&gclid=Cj0KCQiA7NKBBhDBARIsAHbXCB7OLVWs8pYmpW31PIvUl-AppU9qdI8QLsUKJMSsiLifTs41no_v3aYaAhwZEALw_wcB"
              rel="noreferrer"
              target="_blank"
            >
              LinkedIn
            </a>
          </li>
          <div>
            <a
              href="/resources"
              onClick={[Function]}
            >
              <button>
                x
              </button>
            </a>
          </div>
        </div>
        <div>
          <li>
            <a
              href="https://apprentice.at/"
              rel="noreferrer"
              target="_blank"
            >
              Apprentice.at
            </a>
          </li>
          <div>
            <a
              href="/resources"
              onClick={[Function]}
            >
              <button>
                x
              </button>
            </a>
          </div>
        </div>
      </ul>
      <div>
        <a
          href="/newresource"
          onClick={[Function]}
        >
          <button>
            +
          </button>
        </a>
      </div>
    </div>
  </section>
  <hr />
  <section>
    <div>
      <h2>
        Resume & Cover Letter Resources
      </h2>
      <ul>
        <div>
          <li>
            <a
              href="https://resumekraft.com/wp-content/uploads/2019/12/teacher-resume-example.jpg"
              rel="noreferrer"
              target="_blank"
            >
              Resume
            </a>
          </li>
          <div>
            <a
              href="/resources"
              onClick={[Function]}
            >
              <button>
                x
              </button>
            </a>
          </div>
        </div>
        <div>
          <li>
            <a
              href="https://www.thebalancecareers.com/thmb/IC37LV1es8xHWzy2TdJi5sFvIbg=/1650x1275/filters:no_upscale():max_bytes(150000):strip_icc()/The_Balance_Cover_2060250Aug-edddc2451f264a6eb1c94cf627cb77ff.jpg"
              rel="noreferrer"
              target="_blank"
            >
              Cover Letter
            </a>
          </li>
          <div>
            <a
              href="/resources"
              onClick={[Function]}
            >
              <button>
                x
              </button>
            </a>
          </div>
        </div>
        <div>
          <li>
            <a
              href="https://skillsyncer.com/"
              rel="noreferrer"
              target="_blank"
            >
              Resume Optimizer
            </a>
          </li>
          <div>
            <a
              href="/resources"
              onClick={[Function]}
            >
              <button>
                x
              </button>
            </a>
          </div>
        </div>
      </ul>
      <div>
        <a
          href="/newresource"
          onClick={[Function]}
        >
          <button>
            +
          </button>
        </a>
      </div>
    </div>
  </section>
</div>
`;
